packages/webpack-cli/lib/utils/Compiler.js

```diff
@@ -15,17 +16,24 @@ class Compiler {
 
         compilation.hooks.beforeRun.tap('webpackProgress', () => {
             if (outputOptions.progress) {
-                let tmpMsg;
+                process.stdout.write("\n");
                 const defaultProgressPluginHandler = (percent, msg) => {
                     percent = Math.floor(percent * 100);
-                    if (percent === 100) {
-                        msg = 'Compilation completed';
+                    process.stdout.clearLine();
+                    process.stdout.cursorTo(0);
+                    if (percent !== undefined) {
```

Comment:
Will `if (percent)` not work? Are there caveats to that?

Reply:
Zero as percent is a valid value, thus we should print when value is not strictly undefined.

Reply:
Ah cool, my bad. lgtm. 👍